include/boost/beast/http/impl/fields.hpp

```diff
@@ -953,18 +947,22 @@ set_keep_alive_impl(
 template<class Allocator>
 auto
 basic_fields<Allocator>::
-new_element(field name,
-    string_view sname, string_view value) ->
-        element&
-{
-    if(sname.size() + 2 >
-            (std::numeric_limits<off_t>::max)())
-        BOOST_THROW_EXCEPTION(std::length_error{
-            "field name too large"});
-    if(value.size() + 2 >
-            (std::numeric_limits<off_t>::max)())
-        BOOST_THROW_EXCEPTION(std::length_error{
-            "field value too large"});
+new_element(
+    field name,
+    string_view sname,
+    string_view value,
+    error_code& ec) -> element*
+{
+    if(sname.size() > max_name_size)
+    {
+        BOOST_BEAST_ASSIGN_EC(ec, error::header_field_name_too_large);
+        return nullptr;
+    }
+    if(value.size() > max_value_size)
+    {
+        BOOST_BEAST_ASSIGN_EC(ec, error::header_field_value_too_large);
+        return nullptr;
+    }
```

Comment:
Do we also need a limit on the sum of `sname.size()` and `value.size()` (maybe plus delimiters)?

Reply:
technically no, because we use an `unit16_t` for storing the size of value. and another `unit16_t` for its offset. so, name and value can have a size of `std::numeric_limits<std::uint16_t>::max() - 2`.